NetCore/NetCoreRepro/ProxyFactory.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Castle.DynamicProxy;
+using System;
+
+namespace NetCoreRepro
+{
+    public static class ProxyFactory
```

Comment:
Putting all proxy creation inside a centralized class so we don't have to regenerate all test classes all the time. This will come in handy when replacing DynamicProxy with plain System.Reflection.Emit.